client/jetpack-cloud/sections/partner-portal/primary/payment-method-add/index.tsx

```diff
@@ -186,8 +206,12 @@ function PaymentMethodAdd(): ReactElement {
 							<div className="payment-method-add__navigation-buttons">
 								<Button
 									className="payment-method-add__back-button"
-									href="/partner-portal/payment-methods/"
-									disabled={ isStripeLoading }
+									href={
+										product
+											? addQueryArgs( { product }, partnerPortalBasePath( '/issue-license' ) )
+											: '/partner-portal/payment-methods/'
+									}
+									disabled={ isStripeLoading || issueLicense.isLoading || isSubmitting }
```

Comment:
"Go Back" button will lead to the products list with the selected product highlighted, if there's a product selected, keeping the user on the license issuing flow.